src/Particular.HealthMonitoring.Uptime/UptimeMonitoring.cs

```diff
@@ -1,51 +1,41 @@
 ﻿namespace Particular.HealthMonitoring.Uptime
 {
-    using System;
-    using System.Threading;
-    using ServiceControl.Infrastructure;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Particular.HealthMonitoring.Uptime.Api;
     using ServiceControl.Infrastructure.DomainEvents;
-    using ServiceControl.Monitoring;
 
-    public class UptimeMonitoring
+    public class UptimeMonitoringDependencies
     {
-        public EndpointInstanceMonitoring Monitoring { get; }
+        public IPersistEndpointUptimeInformation Persister { get; set; }
+        public IDomainEvents DomainEvents { get; set; }
+    }
 
-        ITimeKeeper timeKeeper;
-        TimeSpan gracePeriod;
-        Timer timer;
+    public class UptimeMonitoring : IComponent<UptimeMonitoringDependencies>
+    {
+        EndpointInstanceMonitoring monitoring;
 
-        public UptimeMonitoring(TimeSpan gracePeriod, IDomainEvents domainEvents, ITimeKeeper timeKeeper)
+        public UptimeMonitoring()
         {
-            this.timeKeeper = timeKeeper;
-            this.gracePeriod = gracePeriod;
-
-            Monitoring = new EndpointInstanceMonitoring(domainEvents);
+            monitoring = new EndpointInstanceMonitoring();
         }
-
-        public void Start()
+        
+        public IEnumerable<object> CreateParts()
         {
-            timer = timeKeeper.New(CheckEndpoints, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+            yield return new UptimeApiModule(monitoring);
+            yield return new HeartbeatFailureDetector(monitoring);
+            yield return new HeartbeatProcessor(monitoring);
+            yield return new EndpointDetectingProcessor(monitoring);
         }
 
-        void CheckEndpoints()
+        public Task Initialize(UptimeMonitoringDependencies dependencies)
         {
-            try
-            {
-                var inactivityThreshold = DateTime.UtcNow - gracePeriod;
-                //log.Debug($"Monitoring Endpoint Instances. Inactivity Threshold = {inactivityThreshold}");
-                Monitoring.CheckEndpoints(inactivityThreshold);
-            }
-            catch (Exception exception)
-            {
-                //log.Error("Exception occurred when monitoring endpoint instances", exception);
-            }
+            return monitoring.Initialize(dependencies.Persister, dependencies.DomainEvents);
         }
 
-        public void Stop()
+        public Task TearDown()
         {
-            timeKeeper.Release(timer);
+            return Task.FromResult(0);
```

Comment:
Shouldn't this chain down to the original timekeeper in `HeartbeatFailureDetector` somehow?

Reply:
The `HeartbeatFailureDetector` needs to be stopped but that's being done by the Startable feature.